src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -770,6 +795,64 @@ class AlterTableColumnStatementSegment(BaseSegment):
     )
 
 
+@snowflake_dialect.segment()
+class AlterWarehouseStatementSegment(BaseSegment):
+    """An `ALTER WAREHOUSE` statement.
+
+    https://docs.snowflake.com/en/sql-reference/sql/alter-warehouse.html
+
+    """
+
+    type = "alter_warehouse_statement"
+    match_grammar = Sequence(
+        "ALTER",
+        "WAREHOUSE",
+        Sequence("IF", "EXISTS", optional=True),
+        OneOf(
+            Sequence(
+                Ref("NakedIdentifierSegment", optional=True),
+                OneOf(
+                    "SUSPEND",
+                    Sequence(
+                        "RESUME",
+                        Sequence("IF", "SUSPENDED", optional=True),
+                    ),
+                ),
+            ),
+            Sequence(
+                Ref("NakedIdentifierSegment", optional=True),
+                Sequence(
+                    "ABORT",
+                    "ALL",
+                    "QUERIES",
+                ),
+            ),
+            Sequence(
+                Ref("NakedIdentifierSegment"),
+                "RENAME",
+                "TO",
+                Ref("NakedIdentifierSegment"),
+            ),
+            Sequence(
+                Ref("NakedIdentifierSegment"),
+                "SET",
+                Sequence(
```

Comment:
Just to clarify -- the 3 things below have to appear in this order?

Reply:
It's difficult to tell to be honest!

The syntax is:
```
ALTER WAREHOUSE [ IF EXISTS ] <name> SET [ objectProperties ]
                                         [ objectParams ]
```

Now Comment is part of `objectProperties` (I've kept it separate so it can be optional, so it doesn't interfere with the create_comments section in the big `CreateSegment` that probably should be split up more!) so really only two segments.

So what's your readying of that? Is order important?

All examples have the Properties before the Params but not sure if that is strictly necessary or not.

This is where having access to a Warehouse database would help!

Still I think this is good enough to merge, and if it turns out order doesn't matter, we can change `Sequence` to `AnyNumberOf`.

Unless you think that's the better option to go with initially?

Reply:
Actually I think it's not important as on different lines, suggesting of equal importance:

```
SET [ objectProperties ]
    [ objectParams ]
```

rather

```
SET [ objectProperties ] [ objectParams ]
```

Will change to `AnyNumberOf`